src/sinks/influxdb/mod.rs

```diff
@@ -171,7 +171,6 @@ fn healthcheck(
 pub(in crate::sinks) fn influx_line_protocol(
     protocol_version: ProtocolVersion,
     measurement: &str,
-    metric_type: &str,
```

Comment:
What do you think about consolidating the insertion of the `metric_type` tag here instead? That is, pass this as `metric_type: Option<&str>` and insert it into the tags if it is `Some`. I think that would reduce the code impact in the callers.

Reply:
From my point of view, the `metric_type` is an implementation detail left by influxdb_metrics sink, which is applicable in a much narrower context. In broader use cases, metric_type is not needed at all. Let's say we would like to sink weather metrics from a station,

```
weather,station=SantaMonica temperature=12,humidity=15%,wind_speed=5km/h <timestamp>
```
It is odd having to specify a metric_type for multi-field metrics here.

Since it is not part of the line protocol, a implementer trying to reuse the `influxdb_line_protocol(measurement, metric_type, tags, fields, timestamp)`, will ask herself: what is this metric_type about? what should I do with it?

That's exactly what happened to `influxdb_logs` sink. The sink had to specify metric_type=logs, even though it is not a **metric**, at all. Why would a log event have to specify **metric** type?

Regarding the code impact, well the change are pretty contained, and should be well covered by unit-test. If I glossed over anything, it should be picked up by the full fledged CI tests. Well, I hope so :joy:

Reply:
Actually, I think this is more valuable part of the PR, compared to the exclude thing. 😬

Reply:
And notice, making it optional does not resolve implementer's doubt either. Since she has to figure out what it is, before she can decide either `None` or `Some`. Another drawback of the approach is, the end user is unable to apply encoding rules to the tag.

By the new approach, the implementer who does need a metric_type should find it easy to add it in tags. For those who does not need it, well, does not see it anywhere.